### ISSUE TITLE
Network Manager applet crashes after (4.0.1 -> 4.0.2) upgrade

### ISSUE THREAD

**Issue by spantaleev** [linuxmint/Cinnamon#8073]
* Cinnamon version (cinnamon --version) - Cinnamon 4.0.2
 * Distribution - Arch
 * Graphics hardware *and* driver used: `lspci -nnk | grep -i vga -A3 | grep 'in use'` says "Kernel driver in use: i915"
 * 64 bit

**Issue**

Since the (4.0.1 -> 4.0.2) upgrade, Cinnamon reports an extension crash on start.

There's nothing interesting in `~/.cinnamon/glass.log`, nor is there a `~/.xsession-errors` file to tell me more.

However, I've noticed the Network Manager applet looking weird - showing an icon as if there's no active connection, even though I was connected to a Wi-Fi network (the "Network Settings" program correctly showed me as connected).

Left-clicking the applet icon didn't do anything.

So I figured I should reload the extension (like [this](https://gist.github.com/hyOzd/35e913db620b3ba086fb)). This made Cinnamon freeze.

**Other information**

I'm running an up-to-date Arch. Relevant other packages are:
- `libnm` = `1.14.5dev+17+gba83251bb-1`
- `networkmanager` = `1.14.5dev+17+gba83251bb-1`

Downgrading Cinnamon to 4.0.1 fixes the issue.

Looking at what's been worked on recently, it looks like #7699 may have been the big change that caused this breakage.

**Comment by leigh123linux** [comment]
You will need to  file the issue against the extension

**Comment by leigh123linux** [comment]
I'm closing this as 'works for me' as you haven't provided any logs to show the error.
Fedora has used this change for nearly **two years** without issue using NetworkManager release versions.
Why should we support Archlinux use of NetworkManager git versions?

**Comment by leigh123linux** [comment]
@eli-schwartz cinnamon needs to be built using libnm-dev files, I don't see any dep listed, did you forget it?

https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/cinnamon#n24

**Comment by spantaleev** [comment]
Where are the logs supposed to be? The aforementioned log (`~/.cinnamon/glass.log`) does not contain anything about any crashes, just stuff like "extension loaded in 112ms".

I've got to ask, why is porting to `libnm` (changing the internals of how network management works in such a major way), something that's a good idea to do in a minor/bugfix release (4.0.1 -> 4.0.2)?
Shouldn't this have happened during (`3.x` -> `4.0`) or (`4.0.x` -> `4.1`)?

**Comment by leigh123linux** [comment]
The nm port was planned for 4.0 initial release but got delayed because LMDE3 was waiting for fixed NetworkManager package.

**Comment by eli-schwartz** [comment]
> @eli-schwartz cinnamon needs to be built using libnm-dev files, I don't see any dep listed, did you forget it?

It's an impossible mistake, "libnm" is one of the pkg-config dependencies checked for in `configure.ac`. :p

Arch doesn't split out development files from the main package, so the dependency is just "libnm", and that is already required by network-manager-applet => networkmanager => libnm.

**Comment by eli-schwartz** [comment]
Reloading the extension `network@cinnamon.org` as per the stated dbus command, works without crashes for me.

**Comment by spantaleev** [comment]
The Network Manager applet worked fine after the 4.0.3 upgrade.

And even though I didn't see the "Cinnamon crashed. Restart?" dialog this time, a crash notification message was still being shown, so I've investigated further.

It's only then that I've noticed that `~/.cinnamon/glass.log` was really old (logs from April 2018). `cinnamon-looking-glass` provided me with up to date logs though. I've updated the Troubleshooting page on the Arch wiki to add a [Debugging](https://wiki.archlinux.org/index.php/Cinnamon#Debugging) section mentioning the `cinnamon-looking-glass` tool, so others would have an easier time finding it.

Anyway, via `cinnamon-looking-glass`, I've found out that it's the [Simple system monitor](https://cinnamon-spices.linuxmint.com/desklets/view/29) desklet extension that was now "crashing" (or rather, "would have crashed if let to work"):

```
[simple-system-monitor@ariel]: Found un-versioned desklet simple-system-monitor@ariel with known usage of imports.gi.NMClient
[simple-system-monitor@ariel]: Error importing desklet.js from simple-system-monitor@ariel
```

Seems like #8050 (which I assume landed in the 4.0.3 release) was doing its job nicely.

As a summary, this is a non-issue now. Thanks for helping out!
